pyproject.toml

```diff
@@ -30,6 +30,17 @@ dependencies = [
     "plum-dispatch>=2.2.1",
 ]
 
+[project.optional-dependencies]
+dev = [
+    "pre-commit>=4.1.0",
+]
+test = [
```

Comment:
Nit: let's merge these two groups into just one `dev` dependency, I don't htink there's a usecase for keeping the two separate?

Reply:
The difference is how many tools, e.g. uv and pixi, treat `dev` versus other dependencies. When doing development they install `dev` automatically but not other extras.

Reply:
Okay. I'm familiar with the special-casing for dev. FWIW I think these probably should still be merged (I'll do this in a follow-up PR) as the test dependencies here are lightweight - but I can see in general that might be a useful separation if the test dependencies are heavy and one prefers to rely on CI instead.